flexget/plugins/api_trakt.py

```diff
@@ -695,67 +691,70 @@ def get_trakt(style=None, title=None, year=None, trakt_id=None, trakt_slug=None,
-                movie_id = movie['movie']['ids']['trakt']
-                cache[movie_id] = movie['movie']
-                cache[movie_id]['collected_at'] = dateutil_parse(movie['collected_at'], ignoretz=True)
     except requests.RequestException as e:
         raise plugin.PluginError('Unable to get data from trakt.tv: %s' % e)
 
 
-def update_watched_cache(style, ident, username=None, account=None):
+def update_watched_cache(style_ident, username=None, account=None):
     if account and not username:
         username = 'me'
-    url = get_api_url('users', username, 'watched', style + 's')
+    url = get_api_url('users', username, 'watched', style_ident)
     session = get_session(account=account)
     try:
         data = session.get(url).json()
         if not data:
             log.warning('No watched data returned from trakt.')
             return
-        cache = ApiTrakt.user_cache[ident]['watched'][style + 's']
-        log.verbose('Received %d records from trakt.tv %s\'s watched history', len(data), username)
-        if style == 'show':
+        cache = get_user_cache(username=username, account=account)['watched'][style_ident]
+        log.verbose('Received %d record(s) from trakt.tv %s\'s watched history', len(data), username)
+        if style_ident == 'movies':
+            for movie in data:
+                movie_id = movie['movie']['ids']['trakt']
+                cache[movie_id] = movie['movie']
+                cache[movie_id]['watched_at'] = dateutil_parse(movie['last_watched_at'], ignoretz=True)
+                cache[movie_id]['plays'] = movie['plays']
+        else:
             for series in data:
                 series_id = series['show']['ids']['trakt']
                 cache[series_id] = series['show']
                 cache[series_id]['seasons'] = series['seasons']
                 cache[series_id]['watched_at'] = dateutil_parse(series['last_watched_at'], ignoretz=True)
                 cache[series_id]['plays'] = series['plays']
-        else:
-            for movie in data:
-                movie_id = movie['movie']['ids']['trakt']
-                cache[movie_id] = movie['movie']
-                cache[movie_id]['watched_at'] = dateutil_parse(movie['last_watched_at'], ignoretz=True)
-                cache[movie_id]['plays'] = movie['plays']
     except requests.RequestException as e:
         raise plugin.PluginError('Unable to get data from trakt.tv: %s' % e)
 
 
-def add_user_to_cache(ident):
-    ApiTrakt.user_cache.setdefault(ident, {}).setdefault('watched', {}).setdefault('shows', {})
-    ApiTrakt.user_cache.setdefault(ident, {}).setdefault('watched', {}).setdefault('movies', {})
-    ApiTrakt.user_cache.setdefault(ident, {}).setdefault('collection', {}).setdefault('shows', {})
-    ApiTrakt.user_cache.setdefault(ident, {}).setdefault('collection', {}).setdefault('movies', {})
+def get_user_cache(username=None, account=None):
+    identifier = str(account) + '|' + str(username)
```

Comment:
Do we need to call str here? If user puts non-ascii in their account name this will crash. perhaps `identifier = '{}|{}'.format(account, username or 'me')`